src/components/Modal.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react"
+import {ModalProps} from "../interface/ModalProps.ts";
+
+const Modal: React.FC<ModalProps> = ({isOpen, selectCard}) => {
+    const cardSelect = (cardName: string) => {
+        selectCard(cardName)
+    }
+
+    return (
+        <div className="modal-dimmed" style={{ display: isOpen ? "flex" : "none" }}>
+            <div className="modal">
+                <div className="flex-center">
+                    <div className="modal-item-container" onClick={() => cardSelect("포코 카드")}>
+                        <div className="modal-item-dot"></div>
+                        <span className="modal-item-name">포코 카드</span>
+                    </div>
+                    <div className="modal-item-container" onClick={() => cardSelect("준 카드")}>
+                        <div className="modal-item-dot"></div>
+                        <span className="modal-item-name">준 카드</span>
+                    </div>
+                    <div className="modal-item-container" onClick={() => cardSelect("현석 카드")}>
+                        <div className="modal-item-dot"></div>
+                        <span className="modal-item-name">현석 카드</span>
+                    </div>
+                    <div className="modal-item-container" onClick={() => cardSelect("윤호 카드")}>
+                        <div className="modal-item-dot"></div>
+                        <span className="modal-item-name">윤호 카드</span>
+                    </div>
```

Comment:
r: array로 렌더링하면 중복 코드를 줄일 수 있어요.

```suggestion
                    {CardTypes.map(({ name }) => (
                      <div className="modal-item-container" onClick={() => cardSelect(name)}>
                        <div className="modal-item-dot"></div>
                        <span className="modal-item-name">{name}</span>
                      </div>
                    )}
```

Reply:
수정했습니다.